PopPUNK/utils.py

```diff
@@ -230,25 +237,38 @@ def qcDistMat(distMat, refList, queryList, a_max):
             Reference labels
         queryList (list)
             Query labels (or refList if self)
+        c_max (float)
+            Maximum core distance to allow
         a_max (float)
             Maximum accessory distance to allow
+        ref_isolate (str)
+            Name of reference from which pruning can occur
 
     Returns:
         passed (bool)
             False if any samples failed
     """
     passed = True
+    to_prune = []
 
     # First check with numpy, which is quicker than iterating over everything
-    if np.any(distMat[:,1] > a_max):
+    if np.any(distMat[:,1] > a_max) or np.any(distMat[:,0] > c_max):
         passed = False
         names = iterDistRows(refList, queryList, refList == queryList)
         for i, (ref, query) in enumerate(names):
-            if distMat[i,1] > a_max:
-                sys.stderr.write("WARNING: Accessory outlier at a=" + str(distMat[i,1]) +
+            if distMat[i,0] > c_max or distMat[i,1] > a_max:
+                sys.stderr.write("WARNING: Outlier at c = " + str(distMat[i,0]) + " a = " + str(distMat[i,1]) +
                                  " 1:" + ref + " 2:" + query + "\n")
-
-    return passed
+                if ref_isolate is not None:
+                    if ref == ref_isolate:
+                        to_prune.append(query)
+                    elif query == ref_isolate:
+                        to_prune.append(ref)
+
```

Comment:
Probably better to put the more common branch first
```suggestion
                if ref_isolate is not None:
                    if ref != ref_isolate:
                        to_prune.append(ref)
                    else:
                        to_prune.append(query)

```

Reply:
Updated in 46a9a9a.